AirCasting/ABConnector/BluetoothManager.swift

```diff
@@ -158,32 +211,55 @@ extension BluetoothManager: CBPeripheralDelegate {
                     peripheral.setNotifyValue(true, for: characteristic)
                     hasSomeCharacteristics = true
                 }
+
+                if characteristic.uuid == DOWNLOAD_FROM_SD_CARD_CHARACTERISTIC_UUID {
+                    peripheral.setNotifyValue(true, for: characteristic)
+                    hasSomeCharacteristics = true
+                }
+
+                if characteristic.uuid == DOWNLOAD_META_DATA_FROM_SD_CARD_CHARACTERISTIC_UUID {
+                    peripheral.setNotifyValue(true, for: characteristic)
+                    hasSomeCharacteristics = true
+                }
             }
         }
         hasSomeCharacteristics ? NotificationCenter.default.post(name: .discoveredCharacteristic, object: nil, userInfo: [AirCastingNotificationKeys.DiscoveredCharacteristic.peripheralUUID : peripheral.identifier]) : nil
     }
-    
+
     func peripheral(_ peripheral: CBPeripheral, didUpdateValueFor characteristic: CBCharacteristic, error: Error?) {
         guard let value = characteristic.value else {
             Log.warning("AirBeam sent measurement without value")
             return
         }
+//        characteristicsMappingLock.lock()
```

Comment:
I think after making the lock recursive it should work just fine. But I think performance here might be hurt because we're unlocking only after triggering all observers. I think we should only obtain them in critical section and after that's done (by copying) we can trigger them outside. But maybe let's dig into this if we actually see some performance related issues